pkg/monitoring/metrics/virt-controller/vmistats_collector.go

```diff
@@ -172,68 +172,88 @@ func getGuestOSInfo(vmi *k6tv1.VirtualMachineInstance) (kernelRelease, machine,
-		instanceType = other
+	vendorName := obj.(*instancetypev1beta1.VirtualMachineInstancetype).Labels[instancetypeVendorLabel]
+	if _, isWhitelisted := whitelistedInstanceTypeVendors[vendorName]; isWhitelisted {
+		instanceType = instancetypeName
+	}
 
-		obj, ok, err := clusterInstanceTypeInformer.GetIndexer().GetByKey(instancetypeName)
-		if err != nil || !ok {
-			return instanceType
-		}
+	return instanceType
+}
 
-		vendorName := obj.(*instancetypev1beta1.VirtualMachineClusterInstancetype).Labels[instancetypeVendorLabel]
-		if _, isWhitelisted := whitelistedInstanceTypeVendors[vendorName]; isWhitelisted {
-			instanceType = instancetypeName
-		}
+func fetchClusterInstancetypeName(instancetypeName string) string {
+	instanceType := other
+
+	obj, ok, err := clusterInstanceTypeInformer.GetIndexer().GetByKey(instancetypeName)
+	if err != nil || !ok {
+		return instanceType
+	}
+
+	vendorName := obj.(*instancetypev1beta1.VirtualMachineClusterInstancetype).Labels[instancetypeVendorLabel]
+	if _, isWhitelisted := whitelistedInstanceTypeVendors[vendorName]; isWhitelisted {
+		instanceType = instancetypeName
 	}
 
 	return instanceType
 }
 
 func getVMIPreference(vmi *k6tv1.VirtualMachineInstance) string {
-	preference := none
-
 	if instancetypeName, ok := vmi.Annotations[k6tv1.PreferenceAnnotation]; ok {
-		preference = other
+		return fetchPreferenceName(instancetypeName)
+	}
 
-		obj, ok, err := preferenceInformer.GetIndexer().GetByKey(instancetypeName)
-		if err != nil || !ok {
-			return preference
-		}
+	if instancetypeName, ok := vmi.Annotations[k6tv1.ClusterPreferenceAnnotation]; ok {
+		return fetchClusterPreferenceName(instancetypeName)
+	}
 
-		vendorName := obj.(*instancetypev1beta1.VirtualMachinePreference).Labels[instancetypeVendorLabel]
-		if _, isWhitelisted := whitelistedInstanceTypeVendors[vendorName]; isWhitelisted {
-			preference = instancetypeName
-		}
+	return none
+}
+
+func fetchPreferenceName(preferenceName string) string {
+	preference := other
+
+	obj, ok, err := preferenceInformer.GetIndexer().GetByKey(preferenceName)
+	if err != nil || !ok {
+		return preference
 	}
 
-	if instancetypeName, ok := vmi.Annotations[k6tv1.ClusterPreferenceAnnotation]; ok {
-		preference = other
+	vendorName := obj.(*instancetypev1beta1.VirtualMachinePreference).Labels[instancetypeVendorLabel]
+	if _, isWhitelisted := whitelistedInstanceTypeVendors[vendorName]; isWhitelisted {
+		preference = preferenceName
+	}
 
-		obj, ok, err := clusterPreferenceInformer.GetIndexer().GetByKey(instancetypeName)
-		if err != nil || !ok {
-			return preference
-		}
+	return preference
+}
 
-		vendorName := obj.(*instancetypev1beta1.VirtualMachineClusterPreference).Labels[instancetypeVendorLabel]
-		if _, isWhitelisted := whitelistedInstanceTypeVendors[vendorName]; isWhitelisted {
-			preference = instancetypeName
-		}
+func fetchClusterPreferenceName(preferenceName string) string {
```

Comment:
`fetchInstancetypeName`, `fetchClusterInstancetypeName`, `fetchPreferenceName` and `fetchClusterPreferenceName` might be candidates to make use of the Object interface?

```
func fetchResourceName(name string, indexer cache.Indexer) string {
	obj, ok, err := indexer.GetByKey(name);
	if err != nil || !ok {
		return other
	}
	
	apiObj, ok := obj.(v1.Object)
	if !ok {
		return other
	}

	vendorName := apiObj.GetLabels()[instancetypeVendorLabel]
	if _, isWhitelisted := whitelistedInstanceTypeVendors[vendorName]; isWhitelisted {
		return name
	}

	return other
}
```

Reply:
awesome :rocket:, ty